code/datums/flock/flock.dm

```diff
@@ -392,6 +392,40 @@
 			if(C)
 				C.images -= I
 
+// naming
+
+/datum/flock/proc/pick_name(flock_type)
+	var/name
+	switch(flock_type)
+		if ("flock")
+			while (!name || (locate(name) in flocks))
+				name = "[pick(consonants_lower)][pick(vowels_lower)].[pick(consonants_lower)][pick(vowels_lower)]"
+		if ("flocktrace")
+			var/trace_found = FALSE
+			while (!trace_found)
```

Comment:
Potentially infinite loop, if the flock is very big (9025 entities). It's unlikely to come up, but it's better to protect against it anyway.